sentry/src/main/java/io/sentry/Attachment.java

```diff
@@ -282,12 +322,12 @@ boolean isAddToTransactions() {
   /**
    * Creates a new View Hierarchy Attachment
    *
-   * @param viewHierarchyBytes the serialized View Hierarchy
+   * @param bytesFactory the serialized View Hierarchy
    * @return the Attachment
    */
-  public static @NotNull Attachment fromViewHierarchy(final byte[] viewHierarchyBytes) {
+  public static @NotNull Attachment fromViewHierarchy(final Callable<byte[]> bytesFactory) {
```

Comment:
M: Since Attachment.fromViewHierarchy is a public API, why don't we allow Attachment.fromViewHierarchy(viewHierarchy) instead of the callable? so the user does not need to handle all the serialization bits of a format they don't understand, but rather creating the ViewHierarchy object, a much user-friendlier API, that's how it is on Dart.

See example:

```
SentryAttachment.fromViewHierarchy(SentryViewHierarchy sentryViewHierarchy)
      : this.fromLoader(
          loader: () => Uint8List.fromList(
              utf8JsonEncoder.convert(sentryViewHierarchy.toJson())),
          filename: 'view-hierarchy.json',
          contentType: 'application/json',
        );
```

Reply:
I agree it would be a more user-friendly API, but `Attachment` is just a POJO holding some data. So if we would provide a `ViewHierarchy` as an argument, `Attachment` would need to take care of serializing and thus pull in quite some dependencies, e.g. by calling `Sentry.getCurrentHub().getOptions().getSerializer().serialize(...)`. I aimed to be consistent with the existing `public static @NotNull Attachment fromScreenshot(final byte[] screenshotBytes) ` signature here as well.
But I'd be happy to improve the API on a next iteration for both ViewHierarchy and Screenshots.

Reply:
@markushi that is just an example, the serialization can happen in the `SentryEnvelopeItem` as it is for many other POJOs.
Doing it later would require a breaking change, the current implementation is different from how it's already being done for other features.
IMO, this would be better addressed now.

Reply:
Alright, now I see - thanks! That's actually pretty much what I initially proposed, but I overlooked the responsibilities of `SentryEnvelopeItem` 😅  I've now delegated the serialization to `SentryEnvelopeItem`, looking better now.